lib/faktory_worker/worker.ex

```diff
@@ -69,9 +127,57 @@ defmodule FaktoryWorker.Worker do
 
   defp schedule_beat(%{worker_state: worker_state} = state)
        when worker_state in @valid_beat_states do
-    beat_ref = Process.send_after(state.worker_server, :beat, state.beat_interval)
+    beat_ref = Process.send_after(self(), :beat, state.beat_interval)
     %{state | beat_ref: beat_ref}
   end
 
   defp schedule_beat(state), do: state
+
+  defp handle_fetch_response({{:ok, :no_content}, conn}, state) do
+    %{state | conn: conn}
+  end
+
+  defp handle_fetch_response({{:ok, job}, conn}, state) do
+    job_supervisor =
+      state.worker_config
+      |> faktory_name()
+      |> FaktoryWorker.JobSupervisor.format_supervisor_name()
+
+    job_ref =
+      Task.Supervisor.async_nolink(
+        job_supervisor,
+        state.worker_module,
+        :perform,
+        job["args"],
+        # make this whatever is set for the 'reserve_for' minus 20 seconds
```

Comment:
is this coming in a subsequent PR?

Reply:
Yes, I was going to quickly follow up with that since it should be a quick job.